### DISCUSSION TITLE
TTS without google

### DISCUSSION THREAD

**Discussion by RZHSSNZDH** [organicmaps/organicmaps#8774] [Q&A] [answered]
My phone's OS is lineageOS and I do not have any google service on it. I think it is why the app's voice guide does not work.
1- Is my detection true?
2- If it is, is there a way to resove this?

**Comment by hb0nd** [comment, marked as answer]
That's partially correct, gapps includes a TTS app by default, while vanilla LineageOS doesn't. There are open alternatives such as [RHVoice](https://f-droid.org/en/packages/com.github.olga_yakovleva.rhvoice.android/) and [eSpeak](https://f-droid.org/en/packages/com.reecedunn.espeak/), but they sound very robotic. 

The best option is to install Google's "[Speech Recognition & Synthesis](https://play.google.com/store/apps/details?id=com.google.android.tts)" app either via a Google play client like [Aurora Store](https://f-droid.org/en/packages/com.aurora.store/), or by downloading the APK on a site like APKPure etc. 
You can disable network access to the app via long press to go to the app info, then disabling "allow network access" in the "mobile data usage" panel. I know it being made by Google feels bad, but it really is the best option IMO, particularly after disabling network access.

**Reply by RZHSSNZDH** [reply]
Thank you! I installed eSpeak. It is a little difficult to understand its voice, but it is better than nothing :)
I am strongly standing against installing google services.

**Reply by Misalf-git** [reply]
I have eSpeak installed as well, on a LineageOS-based ROM + MicroG, but I thought I give it another go because RedAuburn's answer was that thorough.
Google's "Speech Recognition & Synthesis" works without proprietary Google services - I don't know if  MicroG fills the holes in that case though.

**Reply by hb0nd** [reply]
yup, the google tts doesn't rely on Google services (although it may use them when network access is allowed)

**Reply by biodranik** [reply]
Did you read https://organicmaps.app/faq/voice/text-to-speech-android-tts/ ?

**Reply by Misalf-git** [reply]
Good to know.
However, before restricting network access, downloading language specific voice files through it's TTS settings seems to be required.

**Comment by GlytchMeister** [comment]
I've installed RHVoice but I can't install Espeak to do the workaround to get RHVoice to work on a custom degoogled ROM.  I'm on Graphene OS.  Espeak shows as incompatible when I try to install it.

Espeak's latest release is also quite old.

Is there any other workaround for this issue?  Perhaps another, more modern FOSS TTS to replace espeak in this workaround?  Or just a FOSS TTS that just works with no workaround?

**Comment by CrumbNZ** [comment]
Hi all,

Just to throw my 2 cents in, I am using lineageOS on 2 phones Samsung Galaxy S10 and a Samsung Galaxy A21s.
Annoyed by OSMand not having a voice I went down the rabbit hole of TTS on LineageOS.

I found that if I installed EspeakNG from aurora store and tested the voice within in the settings menu after downloading the voice data, TTS would work.

But as you know the voice is not too pleasant.

So I installed Speech recognition and Synthesis from google, from aurora store and went through the settings menu choose it as the TTS engine, downloaded the voice data.

What I noticed is that the play button was greyed out if the Preferred  language was not set right. I had it set to system, but my language is English/ New Zealand. 

Google nor Espeak has a language file for that so I set it to U.S, and it worked. This was after I set Espeak to U.S.
Once it was working wuth Espeak and I could test play it, I simply downloaded a voice I liked - English/ Australia for Google speech and it worked fine, Uninstalled Espeak and no more settings crashes.

OSMand now has voice navigation on both units.

I believe it has something to do with Espeak install, because I tried RHvoice and no good.

Long story short install EspeakNG-configure and test. Then Google speech if you prefer the voices. I use MicroG and have no issues.
Hope this helps.
